pilot/pkg/proxy/envoy/v2/debug.go

```diff
@@ -659,6 +661,15 @@ func writeAllADS(w io.Writer) {
 	fmt.Fprint(w, "]\n")
 }
 
+func (s *DiscoveryServer) ready(w http.ResponseWriter, req *http.Request) {
+	if s.ConfigController != nil {
+		if !s.ConfigController.HasSynced() {
+			w.WriteHeader(503)
```

Comment:
Where is hasSynced implemented ?

Reply:
In k8s config controller - returns k8s 'has synced' state.  I think the others return true by default. 

It means k8s has read all configs  - so requests to pilot will have the needed state.

Note that endpoints and services might not be fully loaded - we would need to check the registry controller.